dm/worker/server.go

```diff
@@ -194,6 +200,41 @@ func (s *Server) checkWorkerStart() *Worker {
 	return s.worker
 }
 
+func (s *Server) stopWorker(sourceID string) *Worker {
+	s.Lock()
+	defer s.Unlock()
+	if s.worker == nil {
+		return nil
+	}
+	if s.worker.cfg.SourceID != sourceID {
+		return nil
+	}
+	w := s.worker
+	s.worker = nil
+	return w
+}
+
+func (s *Server) retryWriteEctd(ops ...clientv3.Op) string {
```

Comment:
How about returning `error` instead of `string`?

Reply:
Emmmm, it is just a warning message.